MetadataExtractor/IO/RandomAccessStream.cs

```diff
@@ -0,0 +1,561 @@
+        /// </remarks>
+        /// <returns>the floating point value</returns>
+        /// <param name="index">position within the data buffer to read first byte</param>
+        /// <param name="IsMotorolaByteOrder">byte order for returning the result</param>
+        /// <param name="isSequential">flag indicating if caller is using sequential access</param>
+        /// <exception cref="BufferBoundsException"/>
+        /// <exception cref="IOException"/>
+        public float GetS15Fixed16(long index, bool IsMotorolaByteOrder, bool isSequential)
+        {
+            var bytes = new byte[4];
+            Read(index, bytes, 0, bytes.Length, isSequential, false);
+
+            if (IsMotorolaByteOrder)
+            {
+                float res = bytes[0] << 8 | bytes[1];
+                var d = bytes[2] << 8 | bytes[3];
+                return (float)(res + d / 65536.0);
+            }
+            else
+            {
+                // this particular branch is untested
+                var d = bytes[1] << 8 | bytes[0];
+                float res = bytes[3] << 8 | bytes[2];
+                return (float)(res + d / 65536.0);
+            }
+        }
+
+        /// <summary>Seeks to an index in the sequence.</summary>
+        /// <remarks>
+        /// Seeks to an index in the sequence. If the sequence can't satisfy the request, exceptions are thrown.
+        /// </remarks>
+        /// <param name="index">position within the data buffer to seek to</param>
+        /// <param name="isSequential">optional for testing to indicate whether sequential access was used.</param>
+        /// <exception cref="BufferBoundsException"/>
+        /// <exception cref="IOException"/>
+        public void Seek(long index, bool isSequential = false)
+        {
+            ValidateIndex((index == 0) ? 0 : (index - 1), 1, isSequential);
+        }
+
+        /// <summary>
+        /// Ensures that the buffered bytes extend to cover the specified index. If not, an attempt is made
+        /// to read to that point.
+        /// </summary>
+        /// <remarks>
+        /// If the stream ends before the point is reached, a <see cref="BufferBoundsException"/> is raised.
+        /// </remarks>
+        /// <param name="index">the index from which the required bytes start</param>
+        /// <param name="bytesRequested">the number of bytes which are required</param>
+        /// <param name="isSequential">flag indicating if caller is using sequential access</param>
+        /// <param name="allowPartial">flag indicating whether count should be enforced when validating the index</param>
+        /// <exception cref="BufferBoundsException">negative index, less than 0 bytes, or too many bytes are requested</exception>
+        /// <exception cref="IOException">if the stream ends before the required number of bytes are acquired</exception>
+        public long ValidateIndex(long index, long bytesRequested, bool isSequential, bool allowPartial = false)
```

Comment:
Document `returns` here. It's not clear from the method name what the return value is for.